src/main/java/seedu/address/logic/commands/people/PeopleReturnedCommand.java

```diff
@@ -0,0 +1,131 @@
+package seedu.address.logic.commands.people;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliPrefix.PEOPLE_COMMAND_TYPE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Debt;
+import seedu.address.model.transaction.Loan;
+import seedu.address.model.transaction.TransactionList;
+
+/**
+ * Records that a debt that the user owe, identified using its displayed index in the Debts table, has been
+ * returned to the person identified using its displayed index from the address book.
+ */
+public class PeopleReturnedCommand extends Command {
+    public static final String COMMAND_WORD = "returned";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Records that you have returned a debt that"
+            + " you owed. Parameters: <person's index> (must be a positive integer) "
+            + " i/[<debt's index> (must be a positive integer)]\n"
+            + "Example: " + PEOPLE_COMMAND_TYPE + " "
+            + COMMAND_WORD + " 4 1";
+
+    public static final String MESSAGE_RETURNED_SUCCESS = "Reduced debt to %1$s by %2$s. You now owe %1$s "
+            + "%3$s.";
+    public static final String MESSAGE_NO_DEBT = "You don't owe %1$s any money :)";
+
+    private final Index targetPersonIndex;
+    private final Index targetDebtIndex;
+
+    public PeopleReturnedCommand(Index targetPersonIndex, Index targetDebtIndex) {
+        this.targetPersonIndex = targetPersonIndex;
+        this.targetDebtIndex = targetDebtIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
This function abit long so maybe good to extract out code e.g. from within the if-else loops